Documentation/policy/language.rst

```diff
@@ -414,6 +419,75 @@ but not CIDR prefix ``10.96.0.0/12``
 
         .. literalinclude:: ../../examples/policies/l3/cidr/cidr.json
 
+.. _DNS based:
+
+DNS based
+---------
+
+``toFQDNs`` simplifies specifying egress policy to IPs of remote, external,
+peers. The DNS lookup for each ``matchName`` is done periodically by
+``cilium-agent`` and the result is used to regenerate endpoint policy. This
+allows tracking changing IPs or sets of IPs that may not be known a priori.
+Despite the naming, the ``matchName`` field does not have to be a
+fully-qualified domain name. In cases where search domains are configured, the
+DNS lookups from ``cilium`` will not be qualified and will utilize the search
+list.
+
+The DNS lookups are repeated with an interval of 5 seconds, and are made for
+A(IPv4) and AAAA(IPv6) addresses. Should a lookup fail, the most recent IP data
+is used instead. An IP change will trigger a regeneration of the ``cilium``
+policy for each endpoint, and the updated IPs can be seen in the response from
+``cilium policy get``. Each update will also increment the per ``cilium-agent``
+policy repository revision.
+
+``toFQDNs`` rules cannot contain any other L3 rules and can support L4/L7
```

Comment:
I think it'd be helpful to have concrete examples of what is / isn't allowed here. For instance, what exactly does "support L4/L7" mean in this context? Examples would clarify this.

Reply:
I tried to make it more clear, without adding actual inline policy. Is this what you had in mind? or actual examples of a bad policy (I'm concerned that those examples won't be clearly marked as bad and people will copy-paste them before the actual example below).

Reply:
Fair enough, I'm not too beholden to my suggestion. We can merge this.

Reply:
Nonetheless, if you read this and think "I don't get it" then it's suboptimal. It's more difficult for me to view this because the code is familiar to me :/ Perhaps a tabbed set of examples, each including a `toPorts` or `HTTP` to illustrate this would work? I don't know how to show the negative examples, however.